test/test.js

```diff
@@ -122,3 +125,64 @@ describe('Web pages ',function(){
     });
 });
 
+describe('Google Calendar API', function() {
+    var path = 'http://localhost:8080/calendar/event'
+    var eventForm =
+    {   title: 'Judgement Day',
+        description: '',
+        location: 'Cheyenne Mountain, Colorado, USA',
+        details: 'Skynet becomes self aware',
+        startDate: '1997-08-29',
+        startHour: '02',
+        startMinute: '14',
+        endDate: '1997-08-29',
+        endHour: '23',
+        endMinute: '59',
+        ssid: 'Artificial Intelligence'
+    }
+    var modifyEvent =
+    '&title='         + eventForm.title           +
+    '&description='   + eventForm.description     +
+    '&location='      + eventForm.location        +
+    '&details='       + eventForm.details         +
+    '&startDate='     + eventForm.startDate       +
+    '&endDate='       + eventForm.endDate         +
+    '&startHour='     + eventForm.startHour       +
+    '&endHour='       + eventForm.endHour         +
+    '&startMinute='   + eventForm.startMinute     +
+    '&endMinute='     + eventForm.endMinute       +
+    '&ssid='           + eventForm.ssid;
+
+    describe('Event insertion.', function() {
+        it ('Creates and inserts an event into the Calendar.', function(done) {
+            request.post({url: path, form: eventForm}, function(error, response, body) {
+                expect(response.statusCode).to.equal(302);
+                done();
+            });
+        });
+    });
+
+    describe('Event retreival, modification, deletion.', function() {
+        var startDate = eventForm.startDate + 'T' + eventForm.startHour + ':' + eventForm.startMinute + ':00-00:00';
+        var endDate = eventForm.endDate + 'T' + eventForm.endHour + ':' + eventForm.endMinute + ':00-00:00';
+        var getEvent = '?startDate=' + startDate + '&endDate=' + endDate;
+        it ('Retreives, modifies, and deletes the inserted event.', function(done) {
```

Comment:
@NikolasHadjipanayi That cannot be done. An event ID is required to delete and modify an event. I can only obtain one using the "get event" test. Local variables cannot be assigned to and persist across both 'it' or 'describe' functions in Mocha by design, so I can't just store the eventID and make subsequent tests use it (I did originally had three tests). Since I must have an event ID to do most of the tests, I opted to embed them this way. 

If I do wish to separate the tests. Then inserting an event will lead to a stack up of duplicate events every time the test is run, modification of an event will require a persistent hardcoded ID which means I need to have an event in the calendar ready. And deleting an event requires I add the event I deleted back before someone makes the next test or it won't be able to delete it and will likely fail. Neither of those were particularly attractive.